[iscyng] iP

iscyng opened the pull request.

iscyng committed:
Added greet and exit, changed duke to phoebe

iscyng committed:
Level1. Echo
added echo and personalisation

iscyng committed:
Level1. Echo
added echo and personalisation

iscyng committed:
Added the X marker to check for tasks done or not done,
Added list storage to store tasks
Had to ask for hint in Level 3 and implemented new Class for taskmanagement

iscyng committed:
Added the X marker to check for tasks done or not done,
Added list storage to store tasks
Had to ask for hint in Level 3 and implemented new Class for taskmanagement

iscyng committed:
Revert "Added the X marker to check for tasks done or not done,"

This reverts commit cd8396a1455a32e99c2cea63bda2dcd2dc440776.

iscyng committed:
Level-4, Added Todo Event Deadline classes and modified the output given different inputs.

A-TextUiTesting
A-CodeQuality

iscyng committed:
A-CodeQuality, commented out lines that were not used in final revision

nidhi-nayak commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,89 @@
+
+
+    private static void markTask(String input) {
+
```
get rid of space here to make your code consistent!

nidhi-nayak commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+public class Task {
```
good job naming all your methods and variables! they are all consistent with the java standard coding documentation.

nidhi-nayak commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,89 @@
+            Task newTask = new Task(taskDescription);
+            tasks[taskCount++] = newTask;
+        } else {
+            System.out.println("NO MORE I SHORT TERM MMR");
```
your chatbot is so funny lol! love it!

nidhi-nayak commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,89 @@
+                System.out.println("added: " + input + "\n");
+            }
+        }
+
```
get rid of space!

nidhi-nayak commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,89 @@
+
+    private static void unmarkTask(String input) {
+
+        int taskIndex = Integer.parseInt(input.substring(7).trim()) - 1;
```
here, i think it would be beneficial to separate the "input.substring(7).trim()" and store it in another variable for readability purposes.

nidhi-nayak reviewed (commented):
Honestly, there was not much readability issues here, your code was in great form! I think standardizing the spaces throughout your code -- but thats standardization issues.  Your chatbot messages made me smile :) Love the sass it gives!

iscyng committed:
A-TextUiTesting implemented text ui testing

iscyng committed:
A-TextUiTesting implemented text ui testing

iscyng committed:
Revert "A-TextUiTesting implemented text ui testing"

This reverts commit 4c4f70670c5f45c4411018fbcc604c6b762f8cb4.

iscyng committed:
Level-5. Did not manage to implement with exception class. Managed to do with conditionals, will try again and recommit the version with exception extension.

iscyng committed:
Level-5. A-Exceptions.

Finally managed to implement exceptions class and further improve on it. Previously would only differ exceptions if there was a space " " after todo, deadline, event. Now the exceptions work the same for both cases

iscyng committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Level-5. A-Exceptions.
  Level-5. Did not manage to implement with exception class. Managed to do with conditionals, will try again and recommit the version with exception extension.
  Revert "A-TextUiTesting implemented text ui testing"
  A-TextUiTesting implemented text ui testing

iscyng committed:
Level-6. A-Collections

Implemented Delete method and changed to use ArrayList for dynamic arrays

iscyng committed:
Level-7. A-Jar. Forgot to split into 2 branches. rerunning and list will result in event not outputting although is saved in phoebe.txt file

iscyng committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Level-7. A-Jar. Forgot to split into 2 branches. rerunning and list will result in event not outputting although is saved in phoebe.txt file
  Level-6. A-Collections

okkhoy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,26 @@
+        super(description);
+        this.from = from;
+        this.to = to;
+//
```
??? remove dead code pls.

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+    private static final ArrayList<Task> tasks = new ArrayList<>();
+    public static void main(String[] args) {
+        loadTasks();
+        String greet = "\n" +
```
Can be made a constant.

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+public class Phoebe {
```
Please add some whitespace to separate two regions of the code.

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+    private static void loadTasks() {
```
Fix all issues related to vertical spacing.

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+    }
+    private static Task parseTaskFromString(String data) {
+        String[] parts = data.split("\\|");
+        // Basic validation
```
This looks like a "note to self" comment

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+    private static Task parseTaskFromString(String data) {
+        String[] parts = data.split("\\|");
+        // Basic validation
+        if (parts.length < 3) return null;
```
coding standard violation in the conditional statement here

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+
+
+
+    private static void addTask(String input) throws PhoebeException {
```
All non-trivial methods need header comments

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+    private static void addTask(String input) throws PhoebeException {
+        input = input.trim();
+        Task newTask = null;
+        if (input.toLowerCase().startsWith("todo")) {
+            String description = input.substring(4).trim();
+            if (description.isEmpty()) {
+                throw new PhoebeException("hello fool how can you todo nothing??????");
+            }
+            newTask = new ToDo(description);
+        } else if (input.toLowerCase().startsWith("deadline")) {
+            String[] parts = input.substring(8).split("/by", 2);
+            if (parts.length < 2 || parts[0].trim().isEmpty()) {
+                throw new PhoebeException("if never tell me /by how I know the deadline??? im ded.");
+            }
+            newTask = new Deadline(parts[0].trim(), parts[1].trim());
+        } else if (input.toLowerCase().startsWith("event")) {
+            String[] parts = input.substring(5).split("/from", 2);
+            if (parts.length < 2 || parts[0].trim().isEmpty()) {
+                throw new PhoebeException("you tell me empty event for what");
+            }
+            String[] timeParts = parts[1].trim().split("/to", 2);
+            if (timeParts.length < 2) {
+                throw new PhoebeException("event no (/from and /to)??? then what time u can go home?");
+            }
+            newTask = new Event(parts[0].trim(), timeParts[0].trim(), timeParts[1].trim());
+        } else {
+            throw new PhoebeException("you dont make any sense");
+        }
```
Consider eliminating the arrow head code here. It requires quite some bit of refactoring,

okkhoy commented on src/main/java/Phoebe.java:
```diff
@@ -0,0 +1,199 @@
+        try {
+            File directory = new File("./data");
+            if (!directory.exists()) {
+                directory.mkdirs(); // make dir
```
redundant comment

okkhoy reviewed (commented):
I hope these review comments help in polishing the iP code before submission.

iscyng committed:
A-MoreOOP

Created Ui Storage Parser and TaskList

iscyng committed:
Level-8, Level-9, A-JavaDoc

iscyng committed:
README.md

iscyng committed:
Updated bug that causes event to not be output when rerun

iscyng committed:
no message

iscyng committed:
Updated ReadMe.md with correct Installation instructions

iscyng committed:
Updated ReadMe.md

iscyng committed:
Updated ReadMe.md with phoebe nametag

iscyng committed:
Updated ReadMe.md with phoebe nametag

iscyng committed:
Merge pull request #1 from iscyng/branch-A-JavaDoc

Branch a java doc
